### ISSUE TITLE
How to add App Insights to Function App?

### ISSUE THREAD

**Issue by Fyzxs** [Azure/azure-powershell#9239]
I hope this is the correct place to ask this.

I'm trying to add App Insights to a Function app and am not finding a command to do so. Note: I suspect one exist - Just can't find it.

I currently create the Application Insights as
```powershell
New-AzApplicationInsights `
            -ResourceGroupName MyResourceGroup `
            -Name MyAppInsights `
            -location westus 
```
I create the Function App as
```powershell
New-AzResource `
            -ResourceGroupName MyResourceGroup `
            -ResourceName myappfunctions `
            -location westus `
            -Kind 'functionapp' `
            -ResourceType 'Microsoft.Web/Sites' `
            -Properties @{} `
            -Force
```

In addition, I'm setting the AppSettings for the FunctionApp that include the InstrumentationKey from the generated AppInsights
```powershell
function Create-FunctionApps-Settings{
    $hashTable = [ordered]@{}
    $hashTable["APPINSIGHTS_INSTRUMENTATIONKEY"] = "$(Get-AppInsights-InstrumentationKey)"
    $hashTable["AzureWebJobsStorage"] = Create-StorageAccount-ConnectionString
    $hashTable["FUNCTIONS_EXTENSION_VERSION"] = "~2"
    $hashTable["FUNCTIONS_WORKER_RUNTIME"] = "dotnet"
    return $hashTable
}

function Get-AppInsights-InstrumentationKey{
    (Get-AzApplicationInsights `
        -ResourceGroupName MyResourceGroup   `
        -Name MyAppInsights).InstrumentationKey
}

Set-AzWebApp `
        -ResourceGroupName MyResourceGroup  `
        -Name myappfunctions `
        -AppSettings $(Create-FunctionApps-Settings)

```

How do I get my FunctionApp to be associated with the App Insights? If my generation of the FunctionApp is wrong, and there's a way that includes setting up App Insights - That's also a great solution.

**Comment by MiYanni** [comment]
@Francisco-Gamino Do you have any insights on how to achieve this?

**Comment by tcsdmaic** [comment]
I am also trying to associate a Web App to an existing Application Insights via a Powershell script.  I did find a blog post about a way to accomplish this that is newer than this issue:  https://www.linkedin.com/pulse/enable-application-insights-azure-web-app-powershell-dries-venter
But it seems like it ought to be way easier and more straight-forward than this.  Especially since at least API Management Service has a way to do it, using New-AzApiManagementLogger (so why not apps?).

**Comment by Francisco-Gamino** [comment]
Hello @Fyzxs and @tcsdmaic -- To add an existing application insights project to a function app, run either 1 or 2:

1)  `Update-AzFunctionApp -Name <functionAppName> -ResourceGroupName <resourceGroupName> -ApplicationInsightsName <projectName>` 
2) `Update-AzFunctionApp -Name <functionAppName> -ResourceGroupName <resourceGroupName> -ApplicationInsightsKey <key>`

The `Update-AzFunctionApp` cmdlet is available in **Az version 4.1.0**. To install it, run `Install-Module -Name Az` . Let me know if you run into any issues.

**Comment by Francisco-Gamino** [comment]
Closing issue given that `Update-AzFunctionApp` supports adding/updating an existing application insights project with a function app.
